File: java/src/io/zihengcat/leetcode/array/MeanOfArrayAfterRemovingSomeElements.java
package io.zihengcat.leetcode.array;

import java.util.Arrays;

/**
 * LeetCode 1619. Mean of Array After Removing Some Elements
 * https://leetcode.com/problems/mean-of-array-after-removing-some-elements/ 
 */
public class MeanOfArrayAfterRemovingSomeElements {
    public static void main(String[] args) {
        // ...
    }
    public double trimMean(int[] arr) {
        if (arr == null || arr.length == 0) {
            return 0;
        }
        Arrays.sort(arr);
        int vLen = arr.length;
        int sIdx = vLen * 5 / 100;
        int eIdx = vLen - sIdx;
        int vSum = 0;
        for (int i = sIdx; i < eIdx; i++) {
            vSum += arr[i];
        }
        return 1.0 * vSum / (eIdx - sIdx);
    }
}

/* EOF */

File: go/array/MeanOfArrayAfterRemovingSomeElements.go
package main

import (
	"sort"
)

/**
 * LeetCode 1619. Mean of Array After Removing Some Elements
 * https://leetcode.com/problems/mean-of-array-after-removing-some-elements/
 */
func trimMean(arr []int) float64 {
	sort.Ints(arr)
	vLen := len(arr)
	sIdx := vLen * 5 / 100
	eIdx := vLen - sIdx
	vSum := 0
	for i := sIdx; i < eIdx; i++ {
		vSum += arr[i]
	}
	return float64(vSum) / float64(eIdx-sIdx)
}

func main() {
	// ...
}

// EOF
